### PR TITLE
SDI-23 Cancel visit nomis endpoint - remove commented out code …

### PR DESCRIPTION
**Pull request by steverendell**
… apply PR comments and add validation and tests

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/service/VisitService.kt [hunk 1]
```diff
@@ -122,11 +130,10 @@ class VisitService(
...
     offenderBooking: OffenderBooking,
   ) {
-    val offenderVisitBalance = offenderVisitBalanceRepository.findById(offenderBooking.bookingId)
+    val offenderVisitBalance = offenderBooking.visitBalance
+    if (offenderVisitBalance != null) {
 
-    if (offenderVisitBalance.isPresent) {
-
-      if (offenderVisitBalance.get().remainingPrivilegedVisitOrders!! > 0) {
+      if (offenderVisitBalance.remainingPrivilegedVisitOrders!! > 0) {
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/service/VisitService.kt:136]: very minor, but pretty certain you don't need `!!` here due to line 134 - the Kotlin complier is clever like that

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/Repository.kt [hunk 2]
```diff
@@ -64,4 +69,15 @@ class Repository(
   fun lookupAgency(id: String): AgencyLocation = agencyLocationRepository.findByIdOrNull(id)!!
   fun delete(offender: Offender) = offenderRepository.deleteById(offender.id)
   fun delete(people: Collection<Person>) = personRepository.deleteAllById(people.map { it.id })
+
+  fun lookupVisit(visitId: Long?): Visit {
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/Repository.kt:73]: I'm guessing the parameter doesn't need to be optional here (causing the !! below)

Same for changeVisitStatus?

**Review by andymarke — APPROVED** [review]
LGTM

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/Repository.kt (hunk 2 above)
Comment (steverendell) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/Repository.kt:73]: I just removed the need for it in the test code, e.g. responsebody.visitId is nullable

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/service/VisitService.kt (hunk 1 above)
Comment (steverendell) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/service/VisitService.kt:136]: yep youre right!

**Pull request merged**